### DISCUSSION TITLE
[@types/superagent] Cannot find name 'Request' error unless "dom" is specified in tsconfig.json

### DISCUSSION THREAD

**Discussion by sean-daley** [DefinitelyTyped/DefinitelyTyped#68398] [Issues with a @types package]
Since upgrading @types/supertest to 6.0.2 which brought in @types/superagent 8.1.3, we are now getting tsc compilation errors:
```
node_modules/@types/superagent/lib/node/response.d.ts:13:26 - error TS2304: Cannot find name 'Request'.

13     constructor(request: Request);
                            ~~~~~~~

node_modules/@types/superagent/lib/node/response.d.ts:34:34 - error TS2304: Cannot find name 'Request'.

34     request: InstanceType<typeof Request>;
                                    ~~~~~~~
```

These problems go away if I enable "dom" in my list of libs in tsconf.json. I'd prefer not to do that unless I really have to though.

Should `lib/node/response.d.ts` be using `request.Request` where `request` would come from `import request = require("../..");` similar to what `lib/node/agent.d.ts` currently does?

[a comment by an automated account is not shown]

**Comment by jakebailey** [comment]
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/68402 may be related?
